tests/phpunit/includes/utils.php

```diff
@@ -378,7 +378,7 @@ function gen_tests_array( $name, $array ) {
 /**
  * Use to create objects by yourself
  */
-class MockClass {}
+class MockClass extends stdClass {}
```

Comment:
@derrabus Because the attribute is expected to be temporary, while `stdClass` is expected to continue to support dynamic properties via highly optimized magic methods, even when support for dynamic properties is removed from PHP Core.

Saves us having to change it again later.

Reply:
@derrabus You may find this write-up/proposal for WordPress handling of dynamic properties on a large scale an interesting read: https://core.trac.wordpress.org/ticket/56034

Reply:
If the attribute was really temporary, extending `stdClass` would make sense indeed. The RFC says

> Classes marked with `#[AllowDynamicProperties]` as well as their children can continue using dynamic properties without deprecation or removal.

My interpretation of that line is that the attribute is meant to stay even beyond PHP 9.

Thanks for the pointer!

Reply:
My assessment is based on a combination between the discussions on the internals mailinglist, the RFC and private conversations with PHP Core devs.

Removing the attribute would require a new RFC, but then again, removing support for dynamic properties would as well, so it is likely those will go together.